src/SSHDebugPS/UI/ViewModels/ContainerViewModel.cs

```diff
@@ -0,0 +1,180 @@
+        public static bool operator ==(ContainerViewModel<T> left, ContainerViewModel<T> right)
+        {
+            if (ReferenceEquals(null, left) || ReferenceEquals(null, right))
+            {
+                return ReferenceEquals(left, right);
+            }
+
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(ContainerViewModel<T> left, ContainerViewModel<T> right)
+        {
+            return !(left == right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is IContainerViewModel instance)
+            {
+                return Equals(instance);
+            }
+            return false;
+        }
+
+        public bool Equals(IContainerViewModel instance)
+        {
+            if (instance is object)
+            {
+                return EqualsInternal(instance);
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return GetHashCodeInternal();
+        }
+
+        #endregion
+
+        #region Helpers
+
+        protected abstract bool EqualsInternal(IContainerViewModel instance);
+        protected abstract int GetHashCodeInternal();
+
+        #endregion
+
+        public string Name => Instance.Name;
+        public string Id => Instance.Id;
+
+        private bool _isExpanded;
+        public bool IsExpanded
+        {
+            get => _isExpanded;
+
+            set
+            {
+                if (_isExpanded != value)
+                {
+                    _isExpanded = value;
+                    OnPropertyChanged(nameof(IsExpanded));
+                }
+            }
+        }
+
+        private bool _isSelected;
+        public bool IsSelected
+        {
+            get
+            {
+                return _isSelected;
+            }
+            set
+            {
+                if (_isSelected != value)
+                {
+                    _isSelected = value;
+                    OnPropertyChanged(nameof(IsSelected));
+                }
+            }
+        }
+    }
+
+    public class DockerContainerViewModel
+        : ContainerViewModel<DockerContainerInstance>
+    {
+        public DockerContainerViewModel(DockerContainerInstance instance)
+            : base(instance)
+        { }
+
+        public string FormattedListOfPorts
+        {
+            get
+            {
+                return string.IsNullOrWhiteSpace(Instance.Ports) ?
+                    UIResources.NoPortsText :
+                    Instance.Ports.Replace(", ", "\r\n");
+            }
+        }
+
+        public override WindowsInput.ICommand ExpandCommand => new ContainerUICommand(Expand, "ExpandDetails", "ExpandToolTip");
+
+        private void Expand(object parameter)
+        {
+            IsExpanded = !IsExpanded;
+        }
+
+        // Gets the first 12 characters and appends an ellipsis
+        public string ShortId { get => Id.Substring(0, 12); }
```

Comment:
>Id.Substring(0, 12); [](start = 39, length = 20)

Any chance `Id` is less than or equal to 12 characters?

Reply:
There shouldn't be but i'll add a guard just in case